src/Microsoft.Extensions.ML/MLContextOptions.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.ML;
+
+namespace Microsoft.Extensions.ML
+{
+    /// <summary>
+    /// Provides an options class for MLContext objects.
+    /// </summary>
+    public class MLContextOptions
```

Comment:
This looks like a wrapper of `MLContext`. Couldn't we directly use `MLContext`?

Reply:
That's a good question. @glennc - would this be possible?

Reply:
Could you hook the logger a different way if you didn't have the PostOptionsConfigure step? I.e how will you register the func that can log with ILogger while you are adding services to the service collection?

Reply:
I am not sure I am following. `MLContextOptions` is a container of `MLContext` and I don't see any new utility functions. Can we replace 
```
    public class PostMLContextOptionsConfiguration : IPostConfigureOptions<MLContextOptions>
```
with
```
    public class PostMLContextOptionsConfiguration : IPostConfigureOptions<MLContext>
```
?

Reply:
I looked into it, and you can't just use `MLContext` directly with `IOptions<T>` because:

```C#
public interface IOptions<out TOptions> where TOptions : class, new()
```
IOptions requires the class to have an empty constructor, which MLContext doesn't.

Also, if we did `IOptions<MLContext>`, we couldn't add new options in the future.

So instead, to solve this, I did the following:

1. Renamed `MLContextOptions` to just `MLOptions`. 
2. Changed the one option currently - the `MLContext` you want to use - to be lazy-loaded, which allows users to supply their own `MLContext`, if they want. So users can do something like:

```C#
            services.Configure<MLOptions>(options =>
            {
                options.MLContext = new MLContext(15);
            });
```

Which allows the user to specify the `seed` if they require it. Or they can configure the `ComponentCatalog` if they want.

I believe this alleviates the concern here. We have an `Options` class, which we can add more options to in the future. And one of those options is which `MLContext` to use.